planner/workers.go

```diff
@@ -12,7 +12,7 @@ import (
 var jobExecutions = make(map[string]*iowrappers.JobExecution)
 
 type Worker interface {
-	handleJob(context.Context, *iowrappers.Job) error
+	handleJob(context.Context, *iowrappers.Job, *sync.RWMutex) error
```

Comment:
why do we need a read write mutex now ? just curious ( for safety ?)

Reply:
just to implement the `worker` interface